### PR TITLE
python3Packages.pyod: init at 1.1.0

### PR DESCRIPTION
**Pull request by mbalatsko**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 259610` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pyod</li>
    <li>python310Packages.pyod.dist</li>
  </ul>
</details>

**Comment by mbalatsko** [comment]
Dependency of https://github.com/NixOS/nixpkgs/issues/247089

File: pkgs/development/python-modules/pyod/default.nix [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    repo = "pyod";
+    rev = "v${version}";
+    hash = "sha256-nPFoN6Zd7kI2dVNGynidGy95n8zudpgLPGznohP7f4Q=";
+  };
+
+  patches = [
+    # Fix ImportError: cannot import name 'DistanceMetric' from 'sklearn.neighbors' error
+    (fetchpatch {
+      name = "fix-sklearn-distancemetric.patch";
+      url = "https://github.com/yzhao062/pyod/pull/513/commits/55ff0d45545da60237642d6d52d086ea4e33cd08.patch";
```
Comment (natsukium) [inline pkgs/development/python-modules/pyod/default.nix:36]: Can we use this commit that is in the master branch? 
https://github.com/yzhao062/pyod/commit/f60813ff48527822c6a865f015a5e60f599116cb

File: pkgs/development/python-modules/pyod/default.nix [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      sha256 = "sha256-kMSYnMvXn2rEIuNHr1GjskF25yyzEh9UrVeSW3xnb0k=";
```
Comment (natsukium) [inline pkgs/development/python-modules/pyod/default.nix:37]:
```suggestion
      hash = "sha256-kMSYnMvXn2rEIuNHr1GjskF25yyzEh9UrVeSW3xnb0k=";
```

File: pkgs/development/python-modules/pyod/default.nix [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    "pyod/test/test_suod.py"
+
+    # pythresh required
+    "pyod/test/test_thresholds.py"
+  ];
+
+  pythonImportsCheck = [ "pyod" ];
+
+  meta = with lib; {
+    description = "A Comprehensive and Scalable Python Library for Outlier Detection (Anomaly Detection";
```
Comment (natsukium) [inline pkgs/development/python-modules/pyod/default.nix:92]:
```suggestion
    description = "A Comprehensive and Scalable Python Library for Outlier Detection (Anomaly Detection)";
```

**Pull request closed without merging**